### PR TITLE
Refactorings and added a ton of logging. Hopefully it helps us understand what is borked. 

### PR DESCRIPTION
**Pull request by CraigHead**

### CODE REVIEW

File: src/Quotebot.Data/Serialization/CosmosSystemTextJsonSerializer.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ public override T FromStream<T>(Stream stream)
             if (buffer is null)
                 throw new NullReferenceException(nameof(buffer));
 
-            if (buffer is T result)
+            if (buffer is { } result)
```
Comment (TheKentuckian) [inline src/Quotebot.Data/Serialization/CosmosSystemTextJsonSerializer.cs:38]: That's a new one!

File: src/Quotebot/Bot.cs [hunk 2]
```diff
@@ -50,18 +49,16 @@ public async Task Connect()
...
+    private async Task LogMessage(LogMessage log)
     {
         _logger.LogDebug(log.Message);
         await Task.CompletedTask;
     }
 
-    public async Task OnReady()
+    private async Task OnReady()
     {
-        await _interactionService.RegisterCommandsToGuildAsync(_configuration.GuildId, true);
```
Comment (TheKentuckian) [inline src/Quotebot/Bot.cs:61]: I'm curious what parameter we were setting to true here. Might be off-topic, but I'm a fan of naming parameters for better readability.

File: src/Quotebot/ServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -29,7 +30,7 @@ public static IServiceCollection RegisterDiscordNet(this IServiceCollection serv
             .AddSingleton<CommandsHandlerService>()
             .AddSingleton<InteractionsHandlerService>()
             .AddSingleton<ItsWednesdayMyDudesService>(serviceProvider =>
-                new(serviceProvider.GetRequiredService<DiscordSocketClient>(), discordConfiguration,
-                    youtubeConfiguration, cancellationTokenSource));
+                new(serviceProvider.GetRequiredService<DiscordSocketClient>(), discordConfiguration, youtubeConfiguration, serviceProvider.GetRequiredService<ILogger<ItsWednesdayMyDudesService>>(), cancellationTokenSource))
```
Comment (TheKentuckian) [inline src/Quotebot/ServiceCollectionExtensions.cs:33]: Nice...

**Review by TheKentuckian — APPROVED** [review]

File: src/Quotebot/Bot.cs [hunk 4]
```diff
@@ -50,18 +49,16 @@ public async Task Connect()
         await _client.StartAsync();
     }
 
-    public async Task LogMessage(LogMessage log)
+    private async Task LogMessage(LogMessage log)
```
[the remark this answers is not in the document]
Comment (CraigHead) [reply src/Quotebot/Bot.cs:52]: @TheKentuckian good question. Without looking and thinking hard, I'm not 100% sure.

**Pull request merged**